### PR TITLE
doc: kernel: improve comments in debug.rs and process.rs

### PR DESCRIPTION
**Pull request by bradjc**
### Pull Request Overview

More kernel doc improvements. This tries to add more detail and description about return values, as well as more links.

### Testing Strategy

travis

### TODO or Help Wanted

n/a

### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [x] Ran `make prepush`.

### CODE REVIEW

File: kernel/src/process.rs [hunk 1]
```diff
@@ -356,41 +357,63 @@ pub trait Process {
     /// return `None` when called.
     fn pending_tasks(&self) -> usize;
 
-    /// Queue a `Task` for the process. This will be added to a per-process
-    /// buffer and executed by the scheduler. `Task`s are some function the app
-    /// should run, for example a upcall or an IPC call.
+    /// Queue a [`Task`] for the process. This will be added to a per-process
+    /// buffer and executed by the scheduler. [`Task`]s are some function the
+    /// app should run, for example a upcall or an IPC call.
```
Comment (lschuermann) [inline kernel/src/process.rs:362]:
```suggestion
    /// process should run, for example a upcall or an IPC call.
```

**Review by lschuermann — COMMENTED** [review]
LGTM, just a nit.

**Review by lschuermann — APPROVED** [review]

**Comment by lschuermann** [comment]
Netlify is not entirely happy:

```
12:09:49 PM: error: unresolved link to `State::StoppedRunning`
12:09:49 PM:    --> kernel/src/process.rs:440:28
12:09:49 PM:     |
12:09:49 PM: 440 |     /// [`StoppedRunning`](State::StoppedRunning) -> [`Running`](State::Running)
12:09:49 PM:     |                            ^^^^^^^^^^^^^^^^^^^^^ the enum `State` has no variant or associated item named `StoppedRunning`
12:09:49 PM:     |
12:09:49 PM:     = note: `-D rustdoc::broken-intra-doc-links` implied by `-D warnings`
12:09:49 PM:     = help: to override `-D warnings` add `#[allow(rustdoc::broken_intra_doc_links)]`
12:09:49 PM: error: unresolved link to `State::StoppedYielded`
12:09:49 PM:    --> kernel/src/process.rs:441:31
12:09:49 PM:     |
12:09:49 PM: 441 |     /// or [`StoppedYielded`](State::StoppedYielded) ->
12:09:49 PM:     |                               ^^^^^^^^^^^^^^^^^^^^^ the enum `State` has no variant or associated item named `StoppedYielded`
```

**Review by alevy — APPROVED** [review]
This is great modulo the build error.

**Comment by alevy** [comment]
@bradjc if you end up not touching this until after my next couple deadlines this week, I'll fix the build errors.

**Review by ppannuto — APPROVED** [review]

**Review by alevy — DISMISSED** [review]

**Review by alevy — APPROVED** [review]

**Comment by alevy** [comment]
@ppannuto @lschuermann fixed netlify

**Review by ppannuto — APPROVED** [review]

**Pull request merged**